lib/page-generate.jsx

```diff
@@ -100,9 +101,10 @@ function run(location, el) {
     we will be using the value there otherwise we will default to
     what was assigned above.
   */
-  supportedLocale = Object.keys(locales).indexOf(navigator.language) !== -1;
-  currentLocale = supportedLocale ? navigator.language : currentLocale;
+  currentLocale = window.location.pathname.split('/')[1];
   messages = locales[currentLocale] || messages;
+  // Load React's data for the current locale, but if it's not available for that specific country, load the general language data
+  addLocaleData(window.ReactIntlLocaleData[currentLocale] || window.ReactIntlLocaleData[currentLocale.split('-')[0]]);
```

Comment:
Ok, I cleaned it up a little but I feel like the loop makes it harder to read and there's no need for multiple locales to load at once, since changing locale hits the server again, plus we need to read the window anyway for messages.